.test/test/internal_pool_config_test.go

```diff
@@ -0,0 +1,84 @@
+/*
+  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+package test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/aws/aws-advanced-go-wrapper/awssql/internal_pool"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewInternalPoolOptions_Defaults(t *testing.T) {
+	config := internal_pool.NewInternalPoolOptions()
+	
+	assert.NotNil(t, config)
+	// Note: Cannot test private fields directly, but we can test the behavior
+}
+
+func TestNewInternalPoolOptions_WithOptions(t *testing.T) {
+	config := internal_pool.NewInternalPoolOptions(
+		internal_pool.WithMaxOpenConns(10),
+		internal_pool.WithMaxIdleConns(5),
+		internal_pool.WithMaxConnLifetime(time.Hour),
+		internal_pool.WithMaxConnIdleTime(30*time.Minute),
+	)
+	
+	assert.NotNil(t, config)
+}
+
+func TestWithMaxOpenConns(t *testing.T) {
+	option := internal_pool.WithMaxOpenConns(10)
+	assert.NotNil(t, option)
+	
+	config := internal_pool.NewInternalPoolOptions(option)
+	assert.NotNil(t, config)
+}
+
+func TestWithMaxIdleConns(t *testing.T) {
+	option := internal_pool.WithMaxIdleConns(5)
+	assert.NotNil(t, option)
+	
+	config := internal_pool.NewInternalPoolOptions(option)
+	assert.NotNil(t, config)
+}
+
+func TestWithMaxConnLifetime(t *testing.T) {
+	option := internal_pool.WithMaxConnLifetime(time.Hour)
+	assert.NotNil(t, option)
+	
+	config := internal_pool.NewInternalPoolOptions(option)
+	assert.NotNil(t, config)
+}
+
+func TestWithMaxConnIdleTime(t *testing.T) {
+	option := internal_pool.WithMaxConnIdleTime(30 * time.Minute)
+	assert.NotNil(t, option)
+	
+	config := internal_pool.NewInternalPoolOptions(option)
+	assert.NotNil(t, config)
+}
+
+func TestMultipleOptions(t *testing.T) {
+	config := internal_pool.NewInternalPoolOptions(
+		internal_pool.WithMaxOpenConns(20),
+		internal_pool.WithMaxIdleConns(10),
+	)
+	
+	assert.NotNil(t, config)
+}
```

Comment:
```suggestion
}

```
Some other files are missing the blank line as well